internetarchive/openlibrary
# Clear cookies when switching accounts

This closes #8005.


This follows the suggestions proposed by Mek in https://github.com/internetarchive/infogami/pull/215#discussion_r1379004747
by creating handlers to call a new function that clears the cookies "pd" and "sfw"

### Technical

This implementation solves the issue mentioned by clearing the cookies using a handler to infogami, overriding infogami `logout` function. It also does this when 'logging in' as another user from the admin panel of `admin/people`

### Testing

(The video below under Screenshot also shows this step)
1. Login as an admin and observe the 'pd' cookie be created through the inspector
2. Logout as an admin and observe the 'pd' cookie has been cleared.
3. Login as admin again and observe that the pd cookie has been created again.
4. Login as another user using `admin/people` and observe that the pd cookie has been cleared.

### Screenshot


I recorded this video showcasing the 'pd' cookie being cleared when the hotfix is applied. Prior to the hotfix application, the pd cookie will never be cleared.

https://github.com/internetarchive/openlibrary/assets/26779639/79aa4738-f57c-496d-a5d0-21e6a592d8d4

### Stakeholders

@mekarpeles proposed the fix concept and suggested this issue to me as a 'Good First Issue' so he would likely want to review my approach prior to a merge.
